File: doc/api-examples/ruby/list_nodes.rb
# SPDX-License-Identifier: CC0-1.0

require "dbus"

sysbus = DBus.system_bus

bc_service = sysbus["org.eclipse.bluechi"]
bc_object = bc_service["/org/eclipse/bluechi"]
bc_interface = bc_object["org.eclipse.bluechi.Manager"]
nodes = bc_interface.ListNodes

puts "BlueChi nodes:"
puts "================"
nodes.each do |node|
  print "Name: ", node[0], "\n"
  print "Path: ", node[1], "\n"
  print "Status: ", node[2], "\n\n"
end

File: doc/api-examples/rust/list-nodes.rs
// Copyright Contributors to the Eclipse BlueChi project
//
// SPDX-License-Identifier: MIT-0

use dbus::blocking::Connection;
use std::time::Duration;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let conn = Connection::new_system()?;

    let bluechi = conn.with_proxy(
        "org.eclipse.bluechi",
        "/org/eclipse/bluechi",
        Duration::from_millis(5000),
    );

    let (nodes,): (Vec<(String, dbus::Path, String, String)>,) =
        bluechi.method_call("org.eclipse.bluechi.Controller", "ListNodes", ())?;

    for (name, _, status) in nodes {
        println!("Node: {}, Status: {}", name, status);
    }

    Ok(())
}
